utils/config.py

```diff
@@ -3,6 +3,25 @@
 import os
 import json
 
+from core.config import load_bot_config_from_file, generate_bot_config
+
+try:
+    base_path = (Path(os.path.abspath(__file__)) / '..' / '..').resolve()
```

Comment:
We're doing this same thing in https://github.com/Splatoon-Stronghold/cas-stronghold-bot/pull/13/files#diff-375fa31174efbd39047f165f7d8bcdcc63da6e732f7d197440bf4dc880d2d6e9R15, perhaps those could be merged into a common function (which is already done in that PR, so perhaps it could be moved to another file and adapted to give path to either `time.json` or `config_data.json`).

(Or, with https://github.com/Splatoon-Stronghold/cas-stronghold-bot/pull/13#issuecomment-2282890288, maybe it could be just used for this and not used for that.)

Reply:
Definitely agree, would prefer if we had a centralized function to get the base path, and then individual files can add on the necessary extensions or create file specific functions.

Reply:
I guess that upcoming function should also be used in https://github.com/Splatoon-Stronghold/cas-stronghold-bot/blob/c6dd5dd6c9c90ca676e6a7a7376a6b38f251956f/utils/start_time.py#L15 then for now, since it's still used there.

Reply:
This was changed to `BOT_CONFIG_PATH` but I think the idea was to rather have something like:

```python
# somewhere in utils
def get_base_config_path():
    return (Path(os.path.abspath(__file__)) /'..' /'..').resolve()
```

And then add `config_data.json` (-> `config.json` per https://github.com/Splatoon-Stronghold/cas-stronghold-bot/pull/14#discussion_r1713059442) or `time.json` to the result of that in the appropriate places.

That said, having a configurable location is not a bad idea 🤔 Since we can just put that in `.env`.

Reply:
I think that, this file (I haven't seen it xD) has config params that should be inside BotConfig